packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -0,0 +1,255 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:pub_semver/pub_semver.dart';
+
+import '../base/common.dart';
+import '../base/context.dart';
+import '../base/file_system.dart';
+import '../base/os.dart';
+import '../base/platform.dart';
+import '../base/process_manager.dart';
+import '../globals.dart';
+import '../ios/plist_utils.dart';
+
+AndroidStudio get androidStudio =>
+    context.putIfAbsent(AndroidStudio, AndroidStudio.latestValid);
+
+// Android Studio layout:
+
+// Linux/Windows:
+// $HOME/.AndroidStudioX.Y/system/.home
+
+// macOS:
+// /Applications/Android Studio.app/Contents/
+// $HOME/Applications/Android Studio.app/Contents/
+
+// $STUDIO_HOME/gradle/gradle-X.Y.Z/bin/gradle
+
+final Version minGradleVersion = new Version(2, 14, 1);
+
+/// Locate Gradle.
+String get gradleExecutable {
+  // See if the user has explicitly configured gradle-dir.
+  String gradleDir = config.getValue('gradle-dir');
+  if (gradleDir != null) {
+    if (fs.isFileSync(gradleDir)) return gradleDir;
```

Comment:
The point here is to make it trivial to read the code flow by scanning the left margin. By having returns indented, you quickly spot where the code flow is non-linear. By avoiding excessive braces, you avoid noise in the left margin.

dartfmt style is significantly less readable than we want for a project like this. We expect people to learn how to use Flutter by examining our code (it happens all the time, just look at the bugs we get filed or the discussion in the gitter channel), so our code is essentially as important as our documentation in terms of teaching our platform to our developers.